src/DynamoCore/Graph/Nodes/CodeBlockUtils.cs

```diff
@@ -265,6 +266,72 @@ internal static IOrderedEnumerable<KeyValuePair<string, int>> GetDefinitionLineI
 
             return locationMap.OrderBy(p => p.Value);
         }
+
+        /// <summary>
+        /// Extracts cleaned individual code expressions as tooltip strings from a full code block.
+        /// </summary>
+        internal static List<string> GetCleanedCodeExpressionsForTooltips(string code)
+        {
+            if (string.IsNullOrWhiteSpace(code))
+                return new List<string>();
+
+            // Filter out full-line comments and remove inline comments
+            var cleanedCode = string.Join(" ",
+                code.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries)
+                    .Where(line => !line.TrimStart().StartsWith("//"))
+                    .Select(line =>
+                    {
+                        var idx = line.IndexOf("//");
+                        return idx >= 0 ? line.Substring(0, idx) : line;
+                    }));
+
+            // Split by semicolon and return trimmed expressions
+            return cleanedCode
+                .Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(expr => expr.Trim())
+                .Where(expr => !string.IsNullOrEmpty(expr))
+                .Select(expr => expr + ";")
+                .ToList();
+        }
+
+        /// <summary>
+        /// Infers a human-readable static type label (e.g., [int], [string]) from an associative AST node.
+        /// </summary>
+        internal static string InferStaticTypeFromNode(AssociativeNode node)
+        {
+            if (node == null) return null;
+
+            switch (node)
+            {
+                case IntNode:
+                    return "[int]";
+                case DoubleNode:
+                    return "[double]";
+                case BooleanNode:
+                    return "[bool]";
+                case StringNode:
+                    return "[string]";
+                case NullNode:
+                    return "[null]";
+                case CharNode:
+                    return "[char]";
+                case ExprListNode:
+                case ArrayNode:
+                    return "[list]";
+                case RangeExprNode:
+                    return "[range]";
+                case IdentifierNode idNode:
+                    return idNode.Value;
+                case IdentifierListNode:
+                    return "[identifier list]";
```

Comment:
I don't believe this is a user-friendly description. I'm confident that many users will not understand this technical wording.

Reply:
makes sense, what will be your suggestion in this case?

Reply:
There's no straightforward answer, unfortunately. An `IdentifierListNode` can be a function call (`DSCore.List.Sort()`), or property invocation (`circle.CenterPoint.X`). It's essentially any complex expression containing a chain (list) of identifiers, where each identifier can either be a class, a variable, a function, or a property. I suppose the ideal solution here would be to break down the identifier list and inspect its rightmost AST to see if it's a function or property (a getter function in DS), then to label it as either a function or property.